internal/impl/nats/auth.go

```diff
@@ -171,3 +193,30 @@ func loadFileContents(filename string, fs *service.FS) ([]byte, error) {
 
 	return io.ReadAll(f)
 }
+
+func saveNkeyFile(content string) (string, error) {
+	err := os.MkdirAll(filepath.Dir(nkeyTempFile), 0755)
```

Comment:
Do not hard code the tmp dir and file name (multiple sources/sinks will get conflict). Use `os.MkdirTemp` instead (it will use the temp dir from system, so it makes sure you code works on all supported OS's, not just Linux):

```go
	dir, err := os.MkdirTemp("", "nats")
	if err != nil {
		// handle err
	}
        nkeyFile := filepath.Join(dir, "nkeys")
        // create the file and write content to the file ...
```

Reply:
is that a random path?

Reply:
we can work around by just delete all temp file for nkeys, there is a rare chance it deleted other nats's nkey fileà